libs/x-content/src/test/java/org/elasticsearch/xcontent/support/filtering/AbstractXContentFilteringTestCase.java

```diff
@@ -355,12 +355,6 @@ private XContentBuilder filter(Builder sample, Set<String> includes, Set<String>
         }
         FilterPath[] excludesFilter = FilterPath.compile(excludes);
         if (excludesFilter != null && Arrays.stream(excludesFilter).anyMatch(FilterPath::hasDoubleWildcard)) {
-            /*
-             * If there are any double wildcard filters the parser based
-             * filtering produced weird invalid json. Just field names
-             * and no objects?! Weird. Anyway, we can't use it.
-             */
-            assertFalse("can't filter on builder with dotted wildcards in exclude", matchFieldNamesWithDots);
```

Comment:
Did you mean to remove this assertion at the same time?

Reply:
Yeah, the comment no longer applies - it now produces valid JSON, just incorrect.

Reply:
Ok I see, this is asserting that other tests don't do something known to be bad. Gotcha.